### ISSUE TITLE
set_baseline bug

### ISSUE THREAD

**Issue by ian-contiamo** [catboost/catboost#716]
**tl;dr** Using `set_baseline` on the training pool makes predictions completely wrong even though the model fits correctly.

Start with the standard simple example:
```python
from catboost import CatBoostRegressor
# Initialize data
cat_features = [0,1,2]
train_data = [["a","b",1,4,5,6],["a","b",4,5,6,7],["c","d",30,40,50,60]]
test_data = [["a","b",2,4,6,8],["a","d",1,4,50,60]]
train_labels = [10,20,30]
# Initialize CatBoostRegressor
model = CatBoostRegressor(iterations=50, learning_rate=1, depth=2)
```

Use a pool:
```python
from catboost import Pool
p = Pool(train_data, train_labels, cat_features)
```

Train the model:
```python
# Fit model
model.fit(p)
# Get predictions
preds = model.predict(test_data)
```

This works as expected (showing the output here):
```
0:	learn: 13.6167789	total: 449us	remaining: 22ms
...
49:	learn: 1.2033391	total: 9.79ms	remaining: 0us
```

Result: `[15.26, 19.40]`

If I want a different starting point however, I set the baseline:
```python
p.set_baseline([100, 100, 100])
model.fit(p)
```

The model still fits properly (it starts from a higher error, but converges easily):
```
0:	learn: 53.7160746	total: 692us	remaining: 33.9ms
...
49:	learn: 0.0042915	total: 12.5ms	remaining: 0us
```

However, the predictions are all wrong: `model.predict(test_data)` returns `[-76.08, -69.90]`
Apparently the baseline interferes with the prediction.

catboost version: 0.11.2
Operating System: Ubuntu 18.04.1 LTS
CPU: Intel Core i7-6700 CPU @ 3.40GHz

**Comment by annaveronika** [comment]
When you are doing predictions, you need to pass baseline values.
If you are using baseline, you are doing the following - you apply first model, and then you apply the second one on top of this one.
You do it during training by specifying starting point - the values of the first model.
In production you need to firstly calculate values of the first model. And then sum them with predictions of the second model

**Comment by ian-contiamo** [comment]
Thanks, that makes sense! May I suggest you update the documentation? It was not clear at all to me...

> The training starts from these values for all input objects instead of starting from zero.

https://catboost.ai/docs/concepts/python-reference_pool_set_baseline.html?lang=en

**Comment by ian-contiamo** [comment]
Finally, is there no easy way to start the regression from a different value than 0? I was trying to address a problem of vanishing gradients at 0. (Of course, I can always offset all my values by a constant, but I was looking for a simpler solution.)

**Comment by david-waterworth** [comment]
@annaveronika to clarify this, in @ian-contiamo's example a constant baseline of 100 was used. So if I understand properly that's equivalent to first fitting a constant model, then starting the training from that model - i.e. `y ~ a + F(X)` where `F(X)` is the fitted catboost model and `a=100` is the constant model?

So again in @ian-contiamo's example to obtain the correct prediction you'd add 100, i.e. `100+model.predict(test_data)`

Am I understanding this correctly?

**Comment by annaveronika** [comment]
> @annaveronika to clarify this, in @ian-contiamo's example a constant baseline of 100 was used. So if I understand properly that's equivalent to first fitting a constant model, then starting the training from that model - i.e. `y ~ a + F(X)` where `F(X)` is the fitted catboost model and `a=100` is the constant model?
> 
> So again in @ian-contiamo's example to obtain the correct prediction you'd add 100, i.e. `100+model.predict(test_data)`
> 
> Am I understanding this correctly?

Yes, you are right.

**Comment by annaveronika** [comment]
> Thanks, that makes sense! May I suggest you update the documentation? It was not clear to me, all it says is:
> 
> > The training starts from these values for all input objects instead of starting from zero.
> 
> https://catboost.ai/docs/concepts/python-reference_pool_set_baseline.html?lang=en

Could you please help with reformulating the docs? What would be a more clear explanation?

**Comment by annaveronika** [comment]
> Finally, is there an easy way to start the regression from a different value than 0? I was trying to address a problem of vanishing gradients at 0. (Of course, I can always offset all my values by a constant, but I was looking for a simpler solution.)

Training from baseline is training the regression from a different value than 0. One thing though - when doing predictions you need to remember that you were starting from other point than 0.

**Comment by annaveronika** [comment]
Since there is no bug, closing the issue

**Comment by ian-contiamo** [comment]
> Could you please help with reformulating the docs? What would be a more clear explanation?

Have you just changed the docs? When I looked three weeks ago there was a full example that showed how to use `set_baseline` from beginning to end. My suggestion would have been to link from the `set_baseline` page to this example, which is self-explanatory. Unfortunately, I can't find it anymore. 

The only result I get from searching the docs is the summary page that only explains how to set the baseline for the input pool:
https://catboost.ai/docs/concepts/python-reference_pool_set_baseline.html

**Comment by annaveronika** [comment]
The example in the docs was incorrect, we have added an updated example recently:
https://catboost.ai/docs/concepts/python-usages-examples.html#baseline
